### DISCUSSION TITLE
Unable to set environment to production

### DISCUSSION THREAD

**Discussion by YunusEmreAlps** [uvdesk/community-skeleton#594] [Q&A] [answered]
Hello,

I hope you're well and everything is good for you.

Unable to set environment to prod

I finished the setup of uvdesk version 1.0.12. After editing the .env file and setting APP_ENV=prod, and clearing cache, the error occurs.

I get an error when I follow the steps in the documentation:

"There are no commands defined in the server namespaces"

What can I do?

Thank you very much.
Best regards,

**Comment by PeopleInside** [comment, marked as answer]
Where you run the command to clean cache? In with folder?
Have you tried to clean cache from the UI in Dashboard? In the footer should be a small icon with a design of a trash.

> I hope you're well and everything is good for you.

Here all fine , I hope you too!

**Reply by YunusEmreAlps** [reply]
I can't even get to the Dashboard screen. APP_ENV only works when I type 'dev'. When I write 'test' and 'prod' it doesn't work.

**Reply by PeopleInside** [reply]
Should be useful have the error you get.
What version of PHP are you using?
How you installed UVdesk: with steps, by SSH?

**Reply by YunusEmreAlps** [reply]
My PHP version is 7.4.30 and Symfony version is 5.4

**Reply by PeopleInside** [reply]
With PHP 7.4 I had issues with the latest UVdesk, also PHP 7.4 will be soon deprecated.
I suggest to try PHP 8.0 or better PHP 8.1

**Reply by YunusEmreAlps** [reply]
Can you give a PHP 8.0 installation guide for UVDesk?

Because open source max 7.3 opensource installation guide.

**Reply by PeopleInside** [reply]
@YunusEmreAlps you can try to following my post here: https://forums.uvdesk.com/topic/1925/installing-and-setup-uvdesk-for-the-first-time

I think the install process should not be different for PHP 8.0 or PHP 8.1

You can also just try to download the zip file at https://www.uvdesk.com/en/opensource/ (download button)
upload on your server.

Then following point 5 of this guide: https://www.uvdesk.com/en/blog/installation-open-source-helpdesk-uvdesk/

After you can also check how to remove public from the URL https://forums.uvdesk.com/topic/1925/installing-and-setup-uvdesk-for-the-first-time/3?_=1661261377572

**Reply by YunusEmreAlps** [reply]
When I type php bin/console server:run terminal.

I'm facing this error:
appuser@host:/data/SmartDash$ php bin/console server:run 0.0.0.0:80
{"message":"Error thrown while running command \"'server:run' '0.0.0.0:80'\". Message: \"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\".\"","context":{"exception":{"class":"Symfony\\Component\\Console\\Exception\\CommandNotFoundException","message":"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\".","code":0,"file":"/data/SmartDash/vendor/symfony/framework-bundle/EventListener/SuggestMissingPackageSubscriber.php:73"},"command":"'server:run' '0.0.0.0:80'","message":"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\"."},"level":500,"level_name":"CRITICAL","channel":"console","datetime":"2022-08-23T13:43:05.367160+00:00","extra":{}}


  There are no commands defined in the "server" namespace.

 You may be looking for a command provided by the "Debug Bundle" which is currently not installed. Try running "composer require symfony/debug-bundle --dev".


What can I need to do for my cloned github project?

**Reply by PeopleInside** [reply]
I asked before but I ask you again as I never read the reply: where you give the command to clear cache?
This should be done in the UVdesk root folder and not from the root user as can give, in my experience issues.
The command should be run by the FTP user.

I suggested to clean cache from the UI so we can be sure you use the right command.
Also you should be secure that php is reached by the only php command. 

For example in Plesk panel you have to use a full php patch to use php command.
This may depend on your server.

**Reply by yildirima** [reply]
We are running clear cache under the directory that exracted from zip file. After extract zip file we run "c:c" replaced APP_ENV with  prod in .env file. We try to run "php bin/console server:run" 


$ php bin/console c:c

 // Clearing the cache for the dev environment with debug false


#php bin/console server:run
{"message":"Error thrown while running command \"'server:run'\". Message: \"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\".\"","context":{"exception":{"class":"Symfony\\Component\\Console\\Exception\\CommandNotFoundException","message":"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\".","code":0,"file":"/data/SmartDash/vendor/symfony/framework-bundle/EventListener/SuggestMissingPackageSubscriber.php:73"},"command":"'server:run'","message":"There are no commands defined in the \"server\" namespace.\n\nYou may be looking for a command provided by the \"Debug Bundle\" which is currently not installed. Try running \"composer require symfony/debug-bundle --dev\"."},"level":500,"level_name":"CRITICAL","channel":"console","datetime":"2022-08-23T13:59:15.510301+00:00","extra":{}}


  There are no commands defined in the "server" namespace.

  You may be looking for a command provided by the "Debug Bundle" which is currently not installed. Try running "composer require symfony/debug-bundle --dev".

**Reply by yildirima** [reply]
We couldn' start the web server at this point. Without starting web service it's not possible to access UI. Please let me know if this profile parameter works on .env file. If we need to change profile from the ZIP file what we need to do after download ZIP.

**Reply by PeopleInside** [reply]
I'm sorry to see you are not able to install UVdesk.
I'm not part of the UVdesk team but I'm just a passionate user who installed and use UVdesk on Ubuntu system.
I give some help when I can on the official community and I try to monitor discussions here and give my free help.

Just with this errors I'm unable to understand what can be wrong.
Maybe a video or some screenshots can help more.

You can wait for a UVdesk staff reply here but this can take more time respect my replies.
If you want you can try to join a chat with me at https://helpdesk.peopleinside.it/it/blog/free-limited-support-on-third-part-software after reading the article.

Maybe the error you get is just saying server:run is not a correct or recognized command.

**Reply by komal-sh-27** [reply]
Let us know which version of uvdesk you have downloaded and where you have downloaded uvdesk project

If you are using zip so after unzip you will see here two folders (folder into a folder), so you will run the clear cache command in the inner child folder

Make a try once **manual setup** of UVdesk.
Check the below link for manual setup from [here](https://github.com/uvdesk/community-skeleton/issues/290#issuecomment-661113642)

If still facing the error please let us know.

**Reply by YunusEmreAlps** [reply]
@Komal-sharma-2712 and @PeopleInside thanks for your help. I solved this problem.

Sometimes I confuse PHP with javascript. I got an error because I tried to run this project like a javascript project. But at the end of the day, I realized what I need to do.

Thanks Again!

**Reply by PeopleInside** [reply]
glad to read you resolved :)

**Comment by PeopleInside** [comment]
If Im not wrong I see you installed UVdesk with Docker.
I dont use Docker so I cannot assist much about this but... I suggest to use PHP 8.0 or PHP 8.1

If you are already using this updated PHP versions and still have the issue I suggest to try install UVdesk with composer or by FTP.
A guide can be found here: https://forums.uvdesk.com/topic/1925/installing-and-setup-uvdesk-for-the-first-time

**Reply by PeopleInside** [reply]
Sorry wrong link, updated my previous message with the correction :)
